### DISCUSSION TITLE
Speed comparison with jszip?

### DISCUSSION THREAD

**Discussion by Fil** [101arrowz/fflate#78] [General]
In this notebook https://observablehq.com/d/b946227d5da6e6fc I try and compare the decoding speed for zip files between jszip and fflate.

The file I want to decompress is a bunch of jpeg, zipped with MacOS Finder's native "archive" menu.

I'm not sure I'm using fflate correctly, since it seems to be 5 to 10 times slower than jszip.

Thank you!

**Comment by 101arrowz** [comment]
I think the problem is that `fflate` decompresses each part of a ZIP with a new thread if you use the async version but does it all in one thread if you use the sync version. You should pick whichever one works best for your type of data. As mentioned in the README, many large files works better async, otherwise the synchronous version tends to perform better. In this case the JPEGs are not that big (roughly all 300kB). When I say "big" I mean dozens to hundreds of megabytes.

When I replace the `fflate` slot with:

```js
fetch(bigfile)
  .then((d) => d.arrayBuffer())
  .then((d) => {
    mutable fflatetimer = performance.now();
    d = new Uint8Array(d);
    return new Promise((resolve, reject) => {
      // unzipSync, not unzip
      const unzipped = fflate.unzipSync(d);
      mutable fflatetimer = performance.now() - mutable fflatetimer;
      resolve(unzipped);
    });
  })
```

It performs 10-20x faster than JSZip. Hope this helps!

**Reply by Fil** [reply]
Fantastic! Thanks a lot for the explanations
